modules/allpass.h

```diff
@@ -0,0 +1,68 @@
+// #Allpass
+// Allpass filter module
+// ## Credits
+// *Original author(s) : Barry Vercoe, John ffitch
+// *Year : 1991
+// *Ported from soundpipe by Ben Sergentanis, May 2020
+
+#pragma once
+#ifndef DSY_ALLPASS_H
+#define DSY_ALLPASS_H
+
+#include <stdint.h>
+#ifdef __cplusplus
+
+#include <math.h>
```

Comment:
I forgot to mention this previously, but we want to avoid having <math.h> implicitly included into the library as this can cause conflicts with other libraries/integrations that may have their own math implementations.

Where possible use equivalent functions/macros from dsp.h or move whatever requires the math.h calls to the cpp file.

Looks like its fminf here. We have `fmin` in dsp.h.